<launch>
   <arg name="worldName" default="example" />
   <arg name="info_output" default="screen" />
 
   <param name="scan_line" type="int" value="64" />

    <!-- if 1, do mapping 10 Hz, if 2, do mapping 5 Hz. Suggest to use 1, it will adjust frequence automaticlly -->
    <param name="mapping_skip_frame" type="int" value="1" />

    <!-- remove too closed points -->
    <param name="minimum_range" type="double" value="0.6"/>

    <param name="mapping_line_resolution" type="double" value="0.4"/>
    <param name="mapping_plane_resolution" type="double" value="0.8"/>

    <remap from="velodyne_points" to="/X1/points"/>

    <node pkg="aloam_velodyne" type="ascanRegistration" name="ascanRegistration" output="$(arg info_output)" />
    <node pkg="aloam_velodyne" type="alaserOdometry" name="alaserOdometry" output="$(arg info_output)" />
    <node pkg="aloam_velodyne" type="alaserMapping" name="alaserMapping" output="$(arg info_output)" />

    <node pkg="tf2_ros" type="static_transform_publisher" name="stp_circuit_caminit" args="0.06 0 0.408 0 0 0 1 map camera_init" output="screen" />
    <node pkg="tf2_ros" type="static_transform_publisher" name="stp_aftmapped_x1" args="-0.06 0 -0.408 0 0 0 1 aft_mapped X1" output="screen" /> 

    <arg name="rviz" default="false" />
    <group if="$(arg rviz)">
    <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find augmented_robot_trackers)/rviz_cfg/sim_aloam_marv.rviz" />
    </group>

</launch>
